src/lib/init.js

```diff
@@ -34,58 +34,61 @@ function readExternalConsentData(config) {
 	});
 }
 
-export function init (configUpdates) {
-	config.update(configUpdates);
-	log.debug('Using configuration:', config);
+export function init (shouldDisplayCmpUI) {
+	return new Promise((resolve, reject) => {
+		(config.getConsentData ? readExternalConsentData(config) : readConsentCookie())
```

Comment:
Please note that below is only suggestion - we may change it later.

This could be moved to "./src/index.js" and then we may use:
```js
Promise.all([
	shouldDisplay(),
	config.getConsentData ? readExternalConsentData(config) : readConsentCookie()
])
```
Thanks to this we won't have to wait for first promise and then sequentially for the second but for both in parallel.

Reply:
Good point, thanks!